pkg/nodediscovery/nodediscovery.go

```diff
@@ -183,19 +183,21 @@ func (n *NodeDiscovery) StartDiscovery(nodeName string, conf Configuration) {
 		go n.createCiliumNodeResource(conf)
 	}
 
-	go func() {
-		<-n.Registered
-		controller.NewManager().UpdateController("propagating local node change to kv-store",
-			controller.ControllerParams{
-				DoFunc: func(ctx context.Context) error {
-					err := n.Registrar.UpdateLocalKeySync(&n.LocalNode)
-					if err != nil {
-						log.WithError(err).Error("Unable to propagate local node change to kvstore")
-					}
-					return err
-				},
-			})
-	}()
+	if option.Config.KVStore != "" {
```

Comment:
This uses different logic from the daemon to determine if the kvstore is enabled, is that OK?

Reply:
oh, I thought the daemon code was also switched out for this

Reply:
I've based on this:
```
        if option.Config.KVStore == "" {
                log.Info("Skipping kvstore configuration")
        } else {
                d.initKVStore()
        }
```

Do you have something specific in mind?